File: lab9/Currency.cs
abstract class Currency
{
    public double Amount { get; set; }

    public Currency(double amount)
    {
        Amount = amount;
    }

    public abstract double ToUAH();
    public abstract void Print();

    public override string ToString()
    {
        return $"{Amount} ({GetType().Name})";
    }
}

File: lab9/Currency.java
abstract class Currency {
    protected double amount;

    public Currency(double amount) {
        this.amount = amount;
    }

    public abstract double toUAH();
    public abstract void print();

    @Override
    public String toString() {
        return amount + " " + this.getClass().getSimpleName();
    }
}
